api/namex/services/name_request/name_request.py

```diff
@@ -250,10 +247,12 @@ def map_draft_attrs(self, name_request):
 
             # Set this to name_request_service_account
             name_request.userId = user_id
-            name_request.entity_type_cd = request_entity
-            name_request.request_action_cd = request_action
             name_request.submittedDate = datetime.utcnow()
-            name_request.requestTypeCd = self.set_request_type(request_entity, request_action)
+            name_request.entity_type_cd = request_entity
+
+            if request_action:
+                name_request.request_action_cd = request_action
+                name_request.requestTypeCd = self.set_request_type(name_request.entity_type_cd, request_action)
```

Comment:
take this out. the vent listener does this already. Only if the front-end supplies the request type should you be settting it here like any other field supplied by the front-end in the payload:
 name_request.requestTypeCd = request_type_cd

Reply:
I will have to confirm that behaviour first. If it's working, without it I should be able to leave it out.